### PR TITLE
 *: add show table regions syntax

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
Add sql syntax to check split table region status.

```sql
SHOW TABLE [table_name] REGIONS

SHOW TABLE [table_name] INDEX [index_name] REGIONS;
```

### What is changed and how it works?
eg
```sql
tidb > create table t (a int, b int, index idx(a));
Query OK, 0 rows affected
Time: 0.077s
tidb > split table t between (0) and (10000000000000000) regions 10;
Query OK, 0 rows affected
Time: 0.046s
tidb > split table t index idx between (0) and (1000000000) regions 4;
Query OK, 0 rows affected
Time: 0.408s
tidb> show table t regions;
+-----------+---------------------------------------+---------------------------------------+-----------+---------------------+------------+
| REGION_ID | START_KEY                             | END_Key                               | LEADER_ID | PEERS               | SCATTERING |
+-----------+---------------------------------------+---------------------------------------+-----------+---------------------+------------+
| 23273     | t_6068_i_1_03800000002cb4178003000000 | t_6068_r_1000000000000000             | 23276     | 23274, 23275, 23276 | 0          |
| 23277     | t_6068_r_1000000000000000             | t_6068_r_2000000000000000             | 23280     | 23280, 23281, 23282 | 0          |
| 23283     | t_6068_r_2000000000000000             | t_6068_r_3000000000000000             | 23284     | 23284, 23285, 23287 | 0          |
| 23288     | t_6068_r_3000000000000000             | t_6068_r_4000000000000000             | 23291     | 23290, 23291, 23292 | 0          |
| 23293     | t_6068_r_4000000000000000             | t_6068_r_5000000000000000             | 23297     | 23294, 23297, 23298 | 0          |
| 23299     | t_6068_r_5000000000000000             | t_6068_r_6000000000000000             | 23301     | 23300, 23301, 23302 | 0          |
| 23303     | t_6068_r_6000000000000000             | t_6068_r_7000000000000000             | 23307     | 23306, 23307, 23308 | 0          |
| 23309     | t_6068_r_7000000000000000             | t_6068_r_8000000000000000             | 23313     | 23310, 23311, 23313 | 0          |
| 23314     | t_6068_r_8000000000000000             | t_6068_r_9000000000000000             | 23318     | 23316, 23317, 23318 | 0          |
| 22571     | t_6068_r_9000000000000000             |                                       | 22573     | 22572, 22573, 22575 | 0          |
| 23325     | t_6068_i_1_                           | t_6068_i_1_03800000000ee6b28003000000 | 23328     | 23326, 23327, 23328 | 0          |
| 23329     | t_6068_i_1_03800000000ee6b28003000000 | t_6068_i_1_03800000001dcd650003000000 | 23332     | 23332, 23333, 23334 | 0          |
| 23335     | t_6068_i_1_03800000001dcd650003000000 | t_6068_i_1_03800000002cb4178003000000 | 23339     | 23336, 23337, 23339 | 0          |
+-----------+---------------------------------------+---------------------------------------+-----------+---------------------+------------+
13 rows in set
Time: 0.013s
tidb> show table t index idx regions;
+-----------+---------------------------------------+---------------------------------------+-----------+---------------------+------------+
| REGION_ID | START_KEY                             | END_Key                               | LEADER_ID | PEERS               | SCATTERING |
+-----------+---------------------------------------+---------------------------------------+-----------+---------------------+------------+
| 23325     | t_6068_i_1_                           | t_6068_i_1_03800000000ee6b28003000000 | 23328     | 23326, 23327, 23328 | 0          |
| 23329     | t_6068_i_1_03800000000ee6b28003000000 | t_6068_i_1_03800000001dcd650003000000 | 23332     | 23332, 23333, 23334 | 0          |
| 23335     | t_6068_i_1_03800000001dcd650003000000 | t_6068_i_1_03800000002cb4178003000000 | 23339     | 23336, 23337, 23339 | 0          |
| 23273     | t_6068_i_1_03800000002cb4178003000000 | t_6068_r_1000000000000000             | 23276     | 23274, 23275, 23276 | 0          |
+-----------+---------------------------------------+---------------------------------------+-----------+---------------------+------------+

```

### Check List 

Tests 

 - Unit test

Code changes

### CODE REVIEW

File: parser.y [hunk 1]
```diff
@@ -6601,6 +6600,21 @@ ShowStmt:
...
+	{
+                $$ = &ast.ShowStmt{
+                        Tp:	ast.ShowRegions,
+			Table:	$3.(*ast.TableName),
+                }
+	}
+|	"SHOW" "TABLE" TableName "INDEX" Identifier "REGIONS"
+	{
+                $$ = &ast.ShowStmt{
+                        Tp:	ast.ShowRegions,
```
Comment (kennytm) [inline parser.y]: indent by tabs.

File: parser.y [hunk 2]
```diff
@@ -6601,6 +6600,21 @@ ShowStmt:
                         User:	$4.(*auth.UserIdentity),
                 }
         }
+|	"SHOW" "TABLE" TableName "REGIONS"
+	{
+                $$ = &ast.ShowStmt{
+                        Tp:	ast.ShowRegions,
```
Comment (kennytm) [inline parser.y]: (ditto)

File: parser.y [hunk 3]
```diff
@@ -6609,6 +6608,21 @@ ShowStmt:
                         User:	$4.(*auth.UserIdentity),
                 }
         }
+|	"SHOW" "TABLE" TableName "REGIONS"
+	{
+                $$ = &ast.ShowStmt{
+			Tp:	ast.ShowRegions,
```
Comment (winkyao) [inline parser.y]: please adjust the indentment.

Comment (crazycs520) [reply parser.y]: done.

**Comment by kennytm** [comment]
Needs to fix parser.go conflict again 🤷

**Comment by tiancaiamao** [comment]
LGTM
Please address conflicts @crazycs520

**Pull request merged**